gson/src/org/immutables/gson/stream/JsonParserReader.java

```diff
@@ -141,7 +144,7 @@ public String nextName() throws IOException {
   public String nextString() throws IOException {
     requirePeek();
     if (!isLenient()) {
-      expect(VALUE_STRING);
+      expectOneOf(VALUE_STRING, VALUE_NUMBER_FLOAT, VALUE_NUMBER_INT);
```

Comment:
Yes, I agree, this changes expected parser behaviour which is questionable. 
At least there are a couple of tests which should help future fixes.